Makefile

```diff
@@ -170,10 +170,17 @@ ci-tunnel: fix-macos-vendor
 ci-clean:
 	docker image prune --all --filter="label=aro-*=true"
 
+.PHONY: pre-deploy-no-aks
+pre-deploy-no-aks:
+	go run -ldflags "-X github.com/Azure/ARO-RP/pkg/util/version.GitCommit=$(VERSION)" ./cmd/aro pre-deploy-no-aks dev-config.yaml ${LOCATION}
+
+.PHONY: pre-deploy-full
+pre-deploy-full:
```

Comment:
looks like this needs `dev-config.yaml` to function. I'd suggest two changes:

1. Remove `.PHONY` on `dev-config.yaml` because `make dev-config.yaml` actually creates a local file named `dev-config.yaml`, and if it exists, we shouldn't regenerate it (that's a bug in our current code tbh)
2. Add `dev-config.yaml` as a required target before this one.

Reply:
>looks like this needs dev-config.yaml to function

I guess that's correct to `pre-deploy-no-aks` as well.

>Remove .PHONY on dev-config.yaml because make dev-config.yaml actually creates a local file named dev-config.yaml, and if it exists, we shouldn't regenerate it (that's a bug in our current code tbh)

I didn't understand why `.PHONY` creates an issue here. I understand the inconvenient consequences of the code but would it be better to change it to a more user-friendly name (simply `dev-config`) with an if clause to verify the file existence prior to creation.

Reply:
`make` was built to compile c programs, which always had artifacts named the same as the target. We can leverage this functionality without additional logic. Here's the simplest example:

```makefile
file:
    touch file
```

And if you execute these commands, this is what you'll see:

```text
$ make file
touch file
$ make file
make: 'file' is up to date.
```

if the artifact of the given name is already present in the local filesystem, `make` will not attempt to "recompile" it.